docs/reference/outlets.md

```diff
@@ -0,0 +1,179 @@
+
+They are conceptually similar to [Stimulus Targets](https://stimulus.hotwired.dev/reference/targets) but with the difference that they reference a Stimulus controller instance plus its associated controller element.
+
+<meta data-controller="callout" data-callout-text-value='data-search-result-outlet=".result"'>
+<meta data-controller="callout" data-callout-text-value='class="result"'>
+
+
+```html
+<div data-controller="search" data-search-result-outlet=".result">
+  ...
+</div>
+
+...
+
+<div id="results">
+  <div class="result" data-controller="result">...</div>
+  <div class="result" data-controller="result">...</div>
+  ...
+</div>
+```
+
+While a **target** is a specifically marked element **within the scope** of its own controller element, an **outlet** can be located **anywhere on the page** and doesn't necessarily have to be within the controller scope.
+
+## Attributes and Names
+
+The `data-search-result-outlet` attribute is called an _outlet attribute_, and its value is a [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) which you can use to refer to other controller elements which should be available as outlets on the _host controller_.
+
+```html
+data-[identifier]-[outlet]-outlet="[selector]"
+```
+
+<meta data-controller="callout" data-callout-text-value='data-search-result-outlet=".result"'>
+
+
+```html
+<div data-controller="search" data-search-result-outlet=".result"></div>
+```
+
+## Definitions
+
+Define controller identifiers in your controller class using the `static outlets` array. This array declares which other controller identifiers can be used as outlets on this controller:
+
+<meta data-controller="callout" data-callout-text-value='static outlets'>
+<meta data-controller="callout" data-callout-text-value='"result"'>
+
+
+```js
+// search_controller.js
+
+export default class extends Controller {
+  static outlets = [ "result" ]
+
+  connect () {
+    this.resultOutlets.forEach(result => ...)
+  }
+}
+```
+
+## Properties
+
+For each outlet defined in the `static outlets` array, Stimulus adds five properties to your controller, where `[name]` corresponds to the outlet's controller identifier:
+
+| Kind | Property name | Return Type | Effect
+| ---- | ------------- | ----------- | -----------
+| Existential | `has[Name]Outlet` | `Boolean` | Tests for presence of a `[name]` outlet
+| Singular | `[name]Outlet` | `Controller` | Returns the `Controller` instance of the first `[name]` outlet or throws an exception if none is present
+| Plural | `[name]Outlets` | `Array<Controller>` | Returns the `Controller` instances of all `[name]` outlets
+| Singular | `[name]OutletElement` | `Element` | Returns the Controller `Element` of the first `[name]` outlet or throws an exception if none is present
+| Plural | `[name]OutletElements` | `Array<Element>` | Returns the Controller `Element`'s of all `[name]` outlets
+
+## Accessing controllers and elements
```

Comment:
```suggestion
## Accessing Controllers and Elements
```

Most of the documentation appears to use APA title case (major words are title case).